pkg/server/server.go

```diff
@@ -378,3 +378,41 @@ func (s *Server) RuntimeHook(ctx context.Context, req *tetragon.RuntimeHookReque
 	}
 	return &tetragon.RuntimeHookResponse{}, nil
 }
+
+func (s *Server) GetLogLevel(ctx context.Context, req *tetragon.GetLogLevelRequest) (*tetragon.GetLogLevelResponse, error) {
+	logger.GetLogger().Debugf("Client requested current log level: %s", logger.GetLogLevel().String())
+	return &tetragon.GetLogLevelResponse{Level: logger.GetLogLevel().String()}, nil
+}
+
+func (s *Server) SetLogLevel(ctx context.Context, req *tetragon.SetLogLevelRequest) (*tetragon.SetLogLevelResponse, error) {
+	currentLogLevel := logger.GetLogLevel()
+	changedLogLevel, err := logrus.ParseLevel(req.GetLevel())
+	if err != nil {
+		return nil, fmt.Errorf("invalid log level: %s", req.GetLevel())
+	}
+
+	// Temporarily set the logger to the lowest level to ensure the message is logged
+	logger.SetLogLevel(logrus.TraceLevel)
+	logger.GetLogger().WithField("request", req).Infof("Log level changed from %s to %s", currentLogLevel, changedLogLevel.String())
+
+	// Set the logger back to the requested level
+	logger.SetLogLevel(changedLogLevel)
+	return &tetragon.SetLogLevelResponse{Level: changedLogLevel.String()}, nil
+}
+
+func (s *Server) ResetLogLevel(ctx context.Context, req *tetragon.ResetLogLevelRequest) (*tetragon.ResetLogLevelResponse, error) {
+	// TODO: Get the original log level with which Tetragon was originally started with in a nicer way than via "option.Config.LogOpts["level"]".
```

Comment:
I would either drop the todo and create a first issue or do this ;)

Reply:
Thanks for the input. Addressed via https://github.com/cilium/tetragon/pull/2643/commits/2b90834c022e35c50dc6515840a3e228d933b31d.